packages/flutter/lib/src/widgets/selectable_region.dart

```diff
@@ -541,6 +541,14 @@ class SelectableRegionState extends State<SelectableRegion> with TextSelectionDe
     }
   }
 
+  void _finalizeSelectableRegionStatus() {
+    if (_selectionStatusNotifier.value != SelectableRegionSelectionStatus.changing) {
+      // Don't finalize the selection again if it is not currently changing.
+      return;
+    }
+    _selectionStatusNotifier.value = SelectableRegionSelectionStatus.finalized;
+  }
```

Comment:
There is still a risk that someone will edit this class in the future and directly set the status to `finalized` without realizing that they should have used this method instead. However, I can't think of a better way to do it right now :shrug:

Reply:
I take that back, you have that covered with the assert in the setter below, nice!